test/unit/test_cache.py

```diff
@@ -405,3 +407,23 @@ def test_clear(self, tmpdir):
         os.unlink(cache_path)
         c["a"] = 1
         assert os.path.exists(cache_path)
+
+    def test_load_with_expired_entries(self, tmpdir):
+        # Test: https://snowflakecomputing.atlassian.net/browse/SNOW-698526
+
+        # create cache first
+        cache_path = os.path.join(tmpdir, "cache.txt")
+        c1 = cache.SFDictFileCache(file_path=cache_path)
+        c1["a"] = 1
+        c1._save()
+
+        # load cache
+        c2 = cache.SFDictFileCache(file_path=cache_path, entry_lifetime=1)
+        c2["b"] = 2
+        c2["c"] = 3
+        # let the cache expire so that when loading again, clearning cache logic will be triggered
+        time.sleep(1)
```

Comment:
Instead of sleeping 1 second use `NO_LIFETIME` and then you won't need to `sleep`.